src/test/tgen/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# Make sure tests in subdirs append to previously collected tests.
+set(ALL_SHADOW_TESTS "${ALL_SHADOW_TESTS}")
+
+add_subdirectory(fixed_duration)
+add_subdirectory(fixed_size)
```

Comment:
Maybe add a README to this tgen directory with a short explanation of what it's doing?